### DISCUSSION TITLE
Should StructuredData be exposed?

### DISCUSSION THREAD

**Discussion by bdukes** [dillonkearns/elm-pages#218] [General]
I'm using https://github.com/dillonkearns/elm-pages/blob/master/examples/docs as an example for a project with blog posts/articles. In the post/article view, the `head` function uses `StructuredData.article` to create the schema.org markup, but the [`StructuredData` module](https://github.com/dillonkearns/elm-pages/blob/master/src/StructuredData.elm) isn't exposed by the elm-pages package. Is the intention that I should be reimplementing only the pieces my application needs, or is that intended to be available for reuse? Thanks!

**Comment by dillonkearns** [comment]
Hello! Good question. That module isn't intended for broader use because schema.org is a very complex spec with a lot of options, and a lot of data types that can be composed together. That module is more of a place to organize the structured data that this project uses.

I've done some experiments to look at what a full structured data API would look like, but it would be an ambitious project and I'm still not sure what the ideal API (or even a viable API) would look like. So not sure if it's a project I'll personally tackle or not, but maybe someone else wants to give it a go.

It would be more of a separate package rather than part of elm-pages directly either way.
